modules/core/shared/src/main/scala/lucuma/core/model/syntax/TrackingOps.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.syntax
+
+import cats.data.NonEmptyList
+import cats.syntax.all.given
+import lucuma.core.math.Coordinates
+import lucuma.core.model.SiderealTracking
+
+import java.time.Instant
+import scala.annotation.targetName
+
+trait ToTrackingOps {
+  extension(trackings: NonEmptyList[SiderealTracking])
+    // We calculate the coordinates at a given time by doing PM
+    // correction of each tracking and then finding the center
+    @targetName("TrackingList_centerOfAt")
+    def centerOfAt(i: Instant): Option[Coordinates] =
+      trackings
+        .map(_.at(i))
+        .sequence
+        .map(nel => Coordinates.centerOf(nel.toList))
```

Comment:
Let me see if I understand. Right now, `Target.Sidereal` has a `SiderealTracking` member which has the data from the `c_sid_*` columns in `t_target`. That `SiderealTracking` has the `at` method on it. Are you suggesting we add a `NonSiderealTracking` member to `Target.NonsiderealTarget` that contains the `c_nsid` fields? ie. Change the structure of `Target.Nonsidereal`?

Reply:
I have no idea of tracking of Nonsidereals will be done? Maybe this would require additional information not in the table yet? And you're just thinking of adding an empty `Tracking` interface to `Target.Nonsidereal` now?

Reply:
Hi sorry, that wasn't clear.  I expect we'll have a `NonsiderealTracking` member at some point and it will have to do the ephemeris lookup from the database.  I'm not suggesting we change `Target.Nonsidereal` now though or that we implement the `at` method for nonsidereals. Just to (perhaps) promote the `at` method to a `Tracking` interface that `SiderealTracking` can implement now.  That way we could provide a `Tracking.centerOf` to match `Coordinates.centerOf`.  For that matter we could skip all of that and just a make `Target.centerOf` that throws an exception for now for `Nonsidereal`s.

Mostly I think I'd just never noticed an extension method that only applies to a particular type of collection like that.  It's probably fine though.

Reply:
I was just thinking that maybe `Target` should have the `at` method, which `Sidereal` could defer to it's `SiderealTracking`, and `Nonsidereal` could throw. It's not clear to me that `Nonsidereal` should have a `Tracking`. The `Tracking` would presumably require data that is directly in `Nonsidereal`? Although I guess we could change that.

Reply:
On second thought, looking at that raises a bunch of questions that aren't easy to answer right now. I think I'll leave this as is if someone will approve it. We can change `ObjectTracking.fromAsterism` at some point when we know more about how we're going to handle non-sidereals.